### PR TITLE
Add server side connections, pending responses metrics

### PR DESCRIPTION
**Pull request by imasahiro**

### CODE REVIEW

**Comment by imasahiro** [comment]
Because currently our service does not shutdown gracefully, want to know about pendings. Do we have any other metrics that is useful to investigate this?

File: core/src/main/java/com/linecorp/armeria/server/GracefulShutdownSupport.java [hunk 1]
```diff
@@ -56,6 +56,11 @@ static GracefulShutdownSupport disabled() {
      */
     public abstract void dec();
 
+    /**
+     * Returns the number of pending requests.
```
Comment (hyangtack) [inline core/src/main/java/com/linecorp/armeria/server/GracefulShutdownSupport.java]: nit: `requests` -> `responses` for consistency?

Comment (imasahiro) [reply core/src/main/java/com/linecorp/armeria/server/GracefulShutdownSupport.java]: Oops, fixed

**Comment by trustin** [comment]
> Because currently our service does not shutdown gracefully, want to know about pendings. Do we have any other metrics that is useful to investigate this?

A gauge for 'time taken since graceful shutdown started'?

**Comment by imasahiro** [comment]
> A gauge for 'time taken since graceful shutdown started'?

Ah, sounds useful to me. Let me keep this PR as is and will open another PR to add the gauge.

**Comment by trustin** [comment]
Thanks, @imasahiro !

**Pull request merged**